### PR TITLE
feat(oauth): oauth.js

### LINKED ISSUE #8: Perform OAuth flow
- TBD

### PR DESCRIPTION
**Pull request by RhnSaxena**
Fixes #8

### CODE REVIEW

File: lib/oauth.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  obj["access_token"] = data["access_token"];
+  obj["expires_in"] = data["expires_in"];
+  obj["onSetTime"] = new Date().getTime();
+
+  let storageObject = {};
+  storageObject["key"] = "application_token";
+  storageObject["value"] = JSON.stringify(obj);
+  Storage_driver(storageObject, "local", "set");
+
+  application_token = obj;
```
Comment (sid22) [inline lib/oauth.js]: Why not just return the obj?

**Comment by RhnSaxena** [comment]
@sid22 please review the changes.

File: lib/oauth.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+          return storeApplicationToken(JSON.parse(result));
+        }
+      })
+    )
+    .catch((error) => {
+      let obj = {};
+      obj["status"] = "Failure";
+      obj["code"] = error.status;
+      obj["message"] = error;
+      return obj;
```
Comment (sid22) [inline lib/oauth.js]:
```
return {
  "status" : "Failure",
  "code" : error.status,
  "message" : error,
}
```

**Review by sid22 — CHANGES_REQUESTED** [review]
Looks good otherwise, have you tested it locally ?

File: lib/oauth.js (hunk 2 above)
Comment (RhnSaxena) [reply lib/oauth.js]: Yes, I have tested it. It's working.

File: lib/oauth.js (hunk 1 above)
Comment (RhnSaxena) [reply lib/oauth.js]: Did it.

**Pull request merged**